### PR TITLE
JID3314-78: Implement Filtering, Sorting, and Pagination to Discrepancy and Statistics Tables

### PR DESCRIPTION
**Pull request by Carson21**
I added @tanstack/react-table and react-icons as packages in order to facilitate the filtering, sorting, pagination and to have icons on the page. Make sure to do "npm i" before "npm run dev" as these are newly added packages.

@waffy1901 I added the function handleStatsDownload that you should use to complete your task that allows a user to download the csv for the statistics. I also did not make the tables a separate component as it would have been harder to implement your [JID3314-113](https://jid3314.atlassian.net/browse/JID3314-113) task.

Here is what the table looks like:
image

### CODE REVIEW

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 1]
```diff
@@ -1,11 +1,168 @@
-import { useState, useEffect } from "react"
+import { useState, useEffect, useMemo } from "react"
+import DebouncedInput from "./DebouncedInput"
+import Filter from "./Filter"
 import config from "../config.json"
+import { FaSortDown, FaSortUp } from "react-icons/fa6"
```
Comment (waffy1901) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:5]: I was going to suggest using Material UI, but this is actually a better design choice!

Comment (Carson21) [reply CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:5]: Yeah I also use material UI icons below, MD. I just couldn't find a good sorting up and down icon from Material design icons

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 2]
```diff
@@ -110,100 +268,311 @@ export default function Report({ reportID }) {
...
+              <DebouncedInput
+                value={discGlobalFilter ?? ""}
+                onChange={(value) => setDiscGlobalFilter(String(value))}
+                className='p-2 font-lg shadow border border-block'
+                placeholder='Search all columns...'
+              />
+              <button
+                type='button'
+                className='bg-blue-400 text-white px-5 py-2 rounded-md hover:bg-blue-600 flex flex-row items-center justify-around gap-2'
+                onClick={handleDownload}
```
Comment (waffy1901) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:449]: A bit of a nit: I think this function should be renamed to handleResultsDownload.

**Review by waffy1901 — APPROVED** [review]
Overall, I see that many changes were made and the UI has been severely augmented in its aesthetic appeal. Great work and keep it up!

**Pull request merged**